### PR TITLE
Issue40 thredds crawler

### PR DESCRIPTION
**Pull request by mortenwh**
#40

### CODE REVIEW

**Comment by mortenwh** [comment]
@akorosov - could you also review this one? Shouldn't take too much time....

File: geospaas/nansat_ingestor/management/commands/ingest_thredds_crawl.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+from geospaas.utils.utils import validate_uri
+from geospaas.nansat_ingestor.models import Dataset as NansatDataset
+
+def crawl(url, **options):
+    if not validate_uri(url):
+        raise ValueError('Invalid url: %s'%url)
+
+    if options['date']:
+        select = ['(.*%s.*\.nc)' %options['date']]
+        import ipdb
```
Comment (akorosov) [inline geospaas/nansat_ingestor/management/commands/ingest_thredds_crawl.py]: Oops?
```suggestion

```

File: geospaas/nansat_ingestor/management/commands/ingest_thredds_crawl.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    return added
+
+class Command(BaseCommand):
+    args = '<url> <select>'
+    help = '''
+        Add metadata of datasets available on thredds/opendap to archive. 
+        
+        Args:
+            <url>: the url to the thredds server
+            <select>: You can select datasets based on their THREDDS ID using
```
Comment (akorosov) [inline geospaas/nansat_ingestor/management/commands/ingest_thredds_crawl.py]: This seems to be outdated?

File: geospaas/nansat_ingestor/management/commands/ingest_thredds_crawl.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            the 'select' parameter.
+            
+        Example: 
+            (1) Find all Sentinel-2A datasets in 2017
+
+            url = http://nbstds.met.no/thredds/catalog/NBS/S2A/catalog.html
+            select = 2017
```
Comment (akorosov) [inline geospaas/nansat_ingestor/management/commands/ingest_thredds_crawl.py]: Should it be --filename or --date?

File: geospaas/nansat_ingestor/management/commands/ingest_thredds_crawl.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+            (2) Find a specific file
+
+            url = http://nbstds.met.no/thredds/catalog/NBS/S2A/catalog.html
+            select = S2A_MSIL1C_20170201T111301_N0204_R137_T32WNS_20170201T111255.nc
```
Comment (akorosov) [inline geospaas/nansat_ingestor/management/commands/ingest_thredds_crawl.py]: Should it be --filename?

**Comment by mortenwh** [comment]
I have problems getting the tests correctly. This needs to be handled before we can merge...

File: geospaas/nansat_ingestor/management/commands/ingest_thredds_crawl.py (hunk 1 above)
Comment (mortenwh) [reply geospaas/nansat_ingestor/management/commands/ingest_thredds_crawl.py]: removed...

File: geospaas/nansat_ingestor/management/commands/ingest_thredds_crawl.py (hunk 2 above)
Comment (mortenwh) [reply geospaas/nansat_ingestor/management/commands/ingest_thredds_crawl.py]: Yes - fixed now

File: geospaas/nansat_ingestor/management/commands/ingest_thredds_crawl.py (hunk 4 above)
Comment (mortenwh) [reply geospaas/nansat_ingestor/management/commands/ingest_thredds_crawl.py]: yes - fixed now

File: geospaas/nansat_ingestor/management/commands/ingest_thredds_crawl.py (hunk 3 above)
Comment (mortenwh) [reply geospaas/nansat_ingestor/management/commands/ingest_thredds_crawl.py]: date - fixed now

**Comment by mortenwh** [comment]
@akorosov - now this should be ready...

**Pull request merged**